utils/lib/Alert.ts

```diff
@@ -81,6 +138,76 @@ class Alert extends Component<QuickstartConfigAlert[], QuickstartAlertInput[]> {
       };
     });
   }
+
+  /**
+   * Static method that gets the alert policy associated with a quickstart and it's current data sources
+   * @returns - object with alert policy ids, required data sources and NGerrors
+   */
+  static async getAlertPolicyRequiredDataSources(quickstart: {
+    name: string;
+    dataSourceIds: string[];
+  }): Promise<
+    | { alertPolicy: AlertPolicyDataSource }
+    | { alertPolicy: null; errors: ErrorOrNerdGraphError[] }
+  > {
+    const { data, errors } = await fetchNRGraphqlResults<
+      AlertPolicyRequiredDataSourcesQueryVariables,
+      AlertPolicyRequiredDataSourcesQueryResults
+    >({
+      queryString: ALERT_POLICY_REQUIRED_DATA_SOURCES_QUERY,
+      variables: { query: `${quickstart.name} alert policy` },
+    });
+
+    const results = data?.actor?.nr1Catalog?.search?.results;
+
+    if (errors) {
+      return { alertPolicy: null, errors };
+    }
+
+    if (results === undefined || results.length === 0) {
+      const error = new Error(
+        `No alert policy for quickstart ${quickstart.name} exists`
+      );
+
+      return { alertPolicy: null, errors: [error] };
+    }
+
+    const alertPoliciesWithUpdatedDataSources = results?.map(
+      (result: AlertPolicy) => {
+        const currDataSourceIds = result.metadata.requiredDataSources.map(
+          (dataSource) => dataSource.id
+        );
+
+        return {
+          id: result.id,
+          dataSourceIds: [
+            ...new Set([...currDataSourceIds, ...quickstart.dataSourceIds]),
```

Comment:
If the quickstart has more than one data source, we have no way of telling if both data sources relate to the alert policy. So in that case, we want to **not** update the alert policy and notify ourselves that we need to do it manually.
We should probably make sure we are doing this on the dashboard side as well.

Reply:
ah okay makes sense! For whatever reason, I assumed that we would just assign multiple data sources to the alert policy if there were multiple.